File: data/derived_cot/rq2_eval/translation/r1/_converted_code/java_to_py/DataStatistics2.py
import math

class DataStatistics2:

    def __init__(self, data):
        self.data = [float(x) for x in data]

    def get_sum(self):
        return sum(self.data)

    def get_min(self):
        if len(self.data) == 0:
            return float('nan')
        return min(self.data)

    def get_max(self):
        if len(self.data) == 0:
            return float('nan')
        return max(self.data)

    def get_variance(self):
        n = len(self.data)
        if n == 0:
            return float('nan')
        mean = self.get_sum() / n
        squared_diffs = [(x - mean) ** 2 for x in self.data]
        return sum(squared_diffs) / n

    def get_std_deviation(self):
        variance = self.get_variance()
        return math.sqrt(variance)

    def get_correlation(self):
        return 1.0

File: data/derived_cot/rq3_prompting/results/translation/r1/py_to_cpp/code_output/DataStatistics2.cpp
#include <vector>
#include <algorithm>
#include <numeric>
#include <cmath>
#include <stdexcept>

class DataStatistics2 {
private:
    std::vector<double> data;

    // Helper: round to two decimal places
    static double round_to_2(double v) {
        return std::round(v * 100.0) / 100.0;
    }

    void check_not_empty() const {
        if (data.empty()) {
            throw std::runtime_error("zero-size array to reduction operation");
        }
    }

public:
    explicit DataStatistics2(const std::vector<double>& input_data)
        : data(input_data) {}

    double get_sum() const {
        return std::accumulate(data.begin(), data.end(), 0.0);
    }

    double get_min() const {
        check_not_empty();
        return *std::min_element(data.begin(), data.end());
    }

    double get_max() const {
        check_not_empty();
        return *std::max_element(data.begin(), data.end());
    }

    double get_variance() const {
        check_not_empty();
        double mean = std::accumulate(data.begin(), data.end(), 0.0) / data.size();
        double sum_sq = 0.0;
        for (double x : data) {
            double diff = x - mean;
            sum_sq += diff * diff;
        }
        return round_to_2(sum_sq / data.size());  // population variance
    }

    double get_std_deviation() const {
        double var = get_variance();
        return round_to_2(std::sqrt(var));
    }

    double get_correlation() const {
        // For a single variable, correlation with itself is always 1.0
        return 1.0;
    }
};